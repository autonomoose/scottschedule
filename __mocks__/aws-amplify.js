export const { Hub } = jest.requireActual('aws-amplify');

const mockDefaultCurrentSession = {
    idToken: { payload: {'custom:tenant': 'test'} },
    accessToken: { payload: {
            username: 'mockuname',
            sub: 'mocksub'
}}};



export const Auth = {
    currentSession: jest.fn(() => Promise.resolve(mockDefaultCurrentSession)),
    currentAuthenticatedUser: jest.fn(() => Promise.resolve({
        username: 'testuser1',
        deleteUser: jest.fn(() => Promise.resolve('deleted')),
    })),
    userAttributes: jest.fn(() => Promise.resolve([
        {Name: 'sub', Value: 'test'},
        {Name: 'email_verified', Value: 'true'},
        {Name: 'email', Value: 'testuser1@test.com'},
        {Name: 'custom:tenant', Value: 'test'},
        {Name: 'phone_number_verified', Value: 'false'},
        {Name: 'phone_number', Value: '111'},
    ])),
    updateUserAttributes: jest.fn(() => Promise.resolve()),
    verifyCurrentUserAttribute: jest.fn(() => Promise.resolve()),
    verifyCurrentUserAttributeSubmit: jest.fn(() => Promise.resolve()),
    signIn: jest.fn(() => Promise.resolve()),
    signOut: jest.fn(() => Promise.resolve()),
    changePassword: jest.fn((currUsr, oldpass, newpass) => Promise.resolve('changed')),
};

export const Storage = {
    configure: jest.fn(() => Promise.resolve('origMockfile')),
    get: jest.fn(() => Promise.resolve('origMockfile')),
    put: jest.fn(() => Promise.resolve('origMockfile')),
    remove: jest.fn(() => Promise.resolve('removed origMockfile')),
    list: jest.fn(() => Promise.resolve([
        {key: "public/", size: 0},
        {key: "public/testlabel1", size: 23470},
        {key: "public/testlabel2", size: 22270},
    ]))
}

// used by layout before every authenticated page
export const mockAPIgqlGetCurrentUser = {
    'data': {
        'getCurrentUser': {
            adisable: null, agroups: null,
            nname: "user3test - legacy primary",
            perm: "standard", userid: "mockgraphql"
}}};
export const API = {
    post: jest.fn(() => Promise.resolve({'message': 'passed'})),
    graphql: jest.fn(() => Promise.resolve(mockAPIgqlGetCurrentUser)),
};

