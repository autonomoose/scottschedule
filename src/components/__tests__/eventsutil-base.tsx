import React from "react";
import { render } from "@testing-library/react";

import { API } from 'aws-amplify';
import DisplayEvents, { fetchEventsDB } from '../../components/eventsutil';
jest.mock('aws-amplify');

const mockCallback = jest.fn();
const eventList = {
  'testevt': {
    descr: 'test event',
    schedRules: ['begin +2,++2,++2',],
  }
};

const mytest = <DisplayEvents tasks={eventList} select={mockCallback} />
describe("eventsutil - base", () => {
  it("renders snapshot correctly", () => {
    const {container} = render(mytest);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("translates simple graphQL into events", async () => {
      const prevAPIgraphql = API.graphql;
      API.graphql = jest.fn(() => Promise.resolve({'data':
        {'listEvents': {items: [
          {
          descr: 'test event',
          evnames: 'testevt!args',
          rules: null,
          },
          {
          descr: null,
          evnames: 'testevt!begin',
          rules: '+2,++2,++2',
          },

        ]}}
      }));

      const newList = await fetchEventsDB();
      expect(newList).toStrictEqual(eventList);
      API.graphql = prevAPIgraphql;
  });
  it("handles graphQL errors", async () => {
      const prevAPIgraphql = API.graphql;
      API.graphql = jest.fn(() => Promise.reject('mockreject'));

      const newList = await fetchEventsDB();
      expect(newList).toStrictEqual({});
      API.graphql = prevAPIgraphql;
  });

});


