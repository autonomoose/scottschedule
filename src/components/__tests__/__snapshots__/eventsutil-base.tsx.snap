// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eventsutil - base renders snapshot correctly 1`] = `
<ul
  class="MuiList-root MuiList-dense css-16vbaq4-MuiList-root"
>
  <div
    class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding MuiListItem-button css-qyxyi3-MuiButtonBase-root-MuiListItem-root"
    role="button"
    tabindex="0"
  >
    testevt
     - 
    test event
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </div>
</ul>
`;
